pkg/api/v1/status/option.go

```diff
@@ -0,0 +1,32 @@
+package status
+
+import "reflect"
+
+// Option is the generic container for any information that needs to be put into the status of the Custom Resource.
+// This is the way to handle some random data that need to be written to status
+type Option interface {
+	Value() interface{}
+}
+
+// GetOption finds the option by its type
+func GetOption(statusOptions []Option, targetOption Option) (Option, bool) {
```

Comment:
yep, I like the idea of using functions instead of data containers 👍 
I think I liked the idea that the "target" (`AtlasProject` etc) has control over what parts of status are updated in the `updateStatus` instead of delegating to some caller but on the other hand there will still be restriction as only the callbacks working with `atlasProject.Status` will be called and not the others.

Don't see the "cons" you've mentioned - we still can move the "specific" options to subpackages, I think generation doesn't work only if the CRD structs are moved to subpackages, it's ok to move some "helper" code there.